### PR TITLE
Avro import

### PR DESCRIPTION
**Pull request by ericzhang6222**
* Please attach **WIP** tag when this PR is still work in progress.
* Please attach **breaking-change** tag if this PR potentially causes other components to fail or changes previous sysl executable's behaviour.

Fixes # .

Changes proposed in this pull request:
- Added arr.ai transformer for Avro to Sysl
- 

Checklist:
- [ ] Added related tests
- [ ] Made corresponding changes to the documentation

### CODE REVIEW

File: pkg/importer/avro/tests/complex_test.arrai [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        tagList <: StringSequenceTagListItemSequence?[union=['StringSequence', 'null', 'tagListItemSequence'], default="[]"]
+        titleMap <: set of string
+        tagMap <: StringSetTagMapItemSet?
+
+    !alias P1:
+        Relationship
+
+    !alias P2:
+        Relationship
+    !type StringMapRecordItem[json_map_key="key"]:
```
Comment (orlade-anz) [inline pkg/importer/avro/tests/complex_test.arrai:68]: Nit: newline

File: pkg/importer/avro/tests/complex_test.arrai [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        key <: string
+        value <: MapRecord
+
+
+    !type MapRecord:
+        id <: string
+
+
+    !type StringtagMapItemItem[json_map_key="key"]:
```
Comment (orlade-anz) [inline pkg/importer/avro/tests/complex_test.arrai:77]: Nit: if this is the name pattern we're going for, capitalise the start of each part (i.e. the T in TagMapItem). Alternatively something like String_tagMapItem_Item
 ```suggestion
    !type StringTagMapItemItem[json_map_key="key"]:
```

File: pkg/importer/avro/tests/complex_test.arrai [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    !alias width2: width
+    # Unions
+
+    !union IntString:
+        int
+        string
+
+    !union StringInt:
+        string
+        int
```
Comment (orlade-anz) [inline pkg/importer/avro/tests/complex_test.arrai:111]: Nit: Are these not the same thing? We shouldn't have both. If they're generated, sort the inner types by name so they end up the same thing.

File: pkg/importer/avro/tests/complex_test.arrai [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+
+    # Aliases
+
+    !alias md5[~fixed, namespace="com.example"]: bytes(16)
+    
+    !alias width[~fixed, namespace="com.example"]: bytes(16)
+    !alias width1: width
+    !alias width2: width
+    # Unions
```
Comment (orlade-anz) [inline pkg/importer/avro/tests/complex_test.arrai:103]: Nit: newline above

File: pkg/importer/avro/tests/complex_test.arrai [hunk 5]
```diff
@@ -0,0 +1,135 @@
+//test.assert.equal(
```
Comment (orlade-anz) [inline pkg/importer/avro/tests/complex_test.arrai:1]: It's hard to recognise that this file is actually a test. Import the fixtures instead, like

```
  //test.assert.equal(//os.file('complex.golden.sysl'), //{./../transformer}("complex.avsc", "Avro", "com.example"))
```

File: pkg/importer/avro/transformToSyslUnion.arrai [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        _: //seq.concat(types >> //str.title(.))
+    }
+;
+
+let printUnion = \union
+    let types = union >> util.getTypeName(.) >> cond . {(type: typeS, ...): typeS, _: .s};
+    $'${'\n'}${util.indent}!union ${//seq.sub("?", "")(combineTypes(union))}:
+    ${
+        types where .@item != 'null' rank (:.@) >> $'${util.indent}${.}'::
+    }\n'
```
Comment (orlade-anz) [inline pkg/importer/avro/transformToSyslUnion.arrai:24]: Nit: space inside the `${...}` is ignore, so inline these one-liners
```suggestion
    ${types where .@item != 'null' rank (:.@) >> $'${util.indent}${.}'::}\n'
```

File: pkg/importer/avro/transformToSyslUnion.arrai [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+;
+
+let t = \union
+    let types = union >> util.getTypeName(.) >> .s;
+    cond //seq.contains(['null'])(types) {
+        true:
```
Comment (orlade-anz) [inline pkg/importer/avro/transformToSyslUnion.arrai:30]: As before

File: pkg/importer/avro/transformer.arrai [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+let rec extraEnums = \schema
+  cond schema {
+    {'type': (s: 'enum'), 'name': name, ...}:
+      {schema},
+    {'type': (s: 'record'), 'name': name, "fields": (a : fields), ...}:
+      //rel.union(fields => extraEnums(.@item("type"))),
+    {'type': type, 'items': items, ...}: # Map or array
+      cond items {
+        (a: types): //rel.union(types => extraEnums(.@item)), # array
+        (:s): extraEnums(items.s), # primitive
```
Comment (orlade-anz) [inline pkg/importer/avro/transformer.arrai:13]: Nit: The `(:s)` is binding `items.s` to a variable named `s`, so you can just use that.
```suggestion
        (:s): extraEnums(s), # primitive
```

File: pkg/importer/avro/transformer.arrai [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+      //rel.union(types => extraEnums(.@item)),  
+    (a: types): # root is json array 
+      //rel.union(types => extraEnums(.@item)),
+  }
+;
+
+# Extra items transformed to Sysl type !type 
+let rec extraTypes = \schema
+  cond schema {
+    {"type": (s: "record"), "name": (s: name), "fields": (a : fields), ...}:
```
Comment (orlade-anz) [inline pkg/importer/avro/transformer.arrai:26]: You don't use `name` here, you can get rid of it.
```suggestion
    {"type": (s: "record"), "fields": (a : fields), ...}:
```

File: pkg/importer/avro/tests/complex_test.arrai [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+    # Types
+
+    !type EventUUID:
+        id <: string
+
+
+    !type Event[namespace="com.example"]:
+        action <: Action
+        header <: Header
+        body <: Party?[union=['Party', 'null']]
```
[the remark this answers is not in the document]
Comment (andrewemeryanz) [reply pkg/importer/avro/tests/complex_test.arrai:17]: It's correct in this instance. In Avro, the order of the union is important. `null` is usually the first item in the union so we only append an attribute to capture the order if this isn't the case.

File: pkg/importer/avro/tests/complex_test.arrai [hunk 11]
```diff
@@ -0,0 +1,135 @@
+//test.assert.equal(
+`##########################################
+##                                      ##
+##  AUTOGENERATED CODE -- DO NOT EDIT!  ##
+##                                      ##
+##########################################
+Avro[avro_spec="1.0", package="com.example"]:
+    # Types
```
Comment (andrewemeryanz) [inline pkg/importer/avro/tests/complex_test.arrai:8]: I don't think these section headers are necessary.

File: pkg/importer/avro/tests/complex_test.arrai [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+        relationshipType <: string[description="This would be the business key for uniquely identifying a relation"]
+        startDate <: string?[description="Relationship start date", default="null"]
+        endDate <: string?[union=['string', 'null'], description="Relationship end date", default="null"]
+        issueDate <: IntString?[union=['int', 'string', 'null'], description="Relationship issue date", default="null"]
+        closeDate <: StringInt?[description="Relationship close date", default="null"]
+        realCloseDate <: StringInt?[union=['string', 'null', 'int'], description="Relationship real close date", default="null"]
+        updateDate <: StringInt[description="Relationship update date", default="null", order="ascending"]
+        updateDate1 <: StringInt[description="Relationship update date", default="null", order="ascending", alias_of="updateDate"]
+        updateDate2 <: StringInt[description="Relationship update date", default="null", order="ascending", alias_of="updateDate"]
+        titleList <: sequence of string?[default="[]"]
```
[the remark this answers is not in the document]
Comment (andrewemeryanz) [reply pkg/importer/avro/tests/complex_test.arrai:58]: Avro types can have no default value or a variety of complex default values depending on the type. We're not inspecting the default values, just noting their value in string form.

**Comment by ericzhang6222** [comment]
Can't merge until #976 is merged.

File: pkg/importer/avro/tests/complex_test.arrai (hunk 3 above)
Comment (andrewemeryanz) [reply pkg/importer/avro/tests/complex_test.arrai:111]: Good question.

The order of the types within the Sysl union is being used to represent the order of the types within Avro union. The `union` attribute is only added if the order can't be derived by the type (and nullable value) alone. 

I'm tempted to go with this suggestion. It will likely add additional `union` attributes in the general case but I think it's a cleaner representation. @ericzhang6222 I'll update specification.

File: pkg/importer/avro/util.arrai [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+            # type can be ['null', 'string'], ['string', 'int'] or ['null', {'type':'record', ...}] etc.
+            # [(s: 'null'), (s: 'string')] to ['null', 'string']
+            combineTypes(typeArray),
+        {'logicalType': logicalTypeName, 'type': typeName, ...}: # Must be before {'type': typeName, ...}
+            transformLogicalTypes(type), 
+        {'type': typeName, ...}:
+            # it is array, map
+            cond typeName.s {
+                'array': 'sequence of ' + cond type {{'items': {'name': name, ...}, ...}: name.s, _: type('items').s},
+                'map': 'set of ' + cond type {{'items': {'name': name, ...}, ...}: 'String' + name.s + 'Item', _: type('items').s},
```
Comment (orlade-anz) [inline pkg/importer/avro/util.arrai:109]:
```suggestion
                'array': 'sequence of ' + cond type('items') {{'name': (:s), ...}: s, (:s): s},
                'map': 'set of ' + cond type('items') {{'name': (:s), ...}: 'String' + s + 'Item', (:s): s},
```

File: pkg/importer/avro/tests/complex_test.arrai (hunk 3 above)
Comment (andrewemeryanz) [reply pkg/importer/avro/tests/complex_test.arrai:111]: @ericzhang6222 Specification updated.

File: pkg/importer/avro/transformToSyslUnion.arrai [hunk 14]
```diff
@@ -0,0 +1,37 @@
+# Transform Avro union to Sysl union
+let util = //{./util};
+
+let combineTypes = \types
+    let fullTypeNames = types >> util.getTypeName(.);
+    let types = fullTypeNames >> .s;
```
[the remark this answers is not in the document]
Comment (ericzhang6222) [reply pkg/importer/avro/transformToSyslUnion.arrai]: Both `fullTypeNames` and `types` are used in following code.

File: pkg/importer/avro/tests/complex_test.arrai (hunk 11 above)
Comment (ericzhang6222) [reply pkg/importer/avro/tests/complex_test.arrai]: This comes from https://github.com/anzx/acceleration/issues/328
```
sysl import required flags:

-i, --input=<SCHEMA> The schema to import
-a, --app-name=<APP-NAME> The name of the Sysl app.
sysl import optional flags:

-p, --package=<PACKAGE> The name of the Sysl package.
-o, --output=<OUT.SYSL> The output filename.
```

File: pkg/importer/avro/tests/complex_test.arrai [hunk 15]
```diff
@@ -0,0 +1,135 @@
...
+    !type Event[namespace="com.example"]:
+        action <: Action
+        header <: Header
+        body <: Party?[union=['Party', 'null']]
+        mapRecords <: set of StringMapRecordItem
+        relationships <: sequence of Relationship
+        tagList <: sequence of string[default="['tag1', 'tag2']"]
+        len <: md5
+        width <: width
+        occurrenceDateTime <: datetime[~timestamp-millis, description="The date and time, to millisecond precision, that this event occurred"]
```
[the remark this answers is not in the document]
Comment (ericzhang6222) [reply pkg/importer/avro/tests/complex_test.arrai]: @andrewemeryanz do you want updating to `@description` syntax?

File: pkg/importer/avro/transformToSyslType.arrai [hunk 16]
```diff
@@ -0,0 +1,100 @@
...
+    cond type {
+        (a: typeArray):
+            let types = (typeArray >> util.getTypeName(.)) >> .s;
+            cond //seq.contains(['null'])(types) {
+                true:
+                    cond //seq.has_prefix(['null'])(types) {
+                        true: # types is ['null', 'string']
+                            '',
+                        _: # types is ['string', 'null']
+                            $'union=${types}'
```
[the remark this answers is not in the document]
Comment (ericzhang6222) [reply pkg/importer/avro/transformToSyslType.arrai]: If types is `['null', 'string']`, it outputs code `attr1 <: string?`. Or outputs code `attr <: string? [union=['string', 'null']]`.
https://confluence.service.anz/display/ABT/Avro+representation#Avrorepresentation-unions

File: pkg/importer/avro.go [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+	logger  *logrus.Logger
+}
+
+// Load returns Sysl result
+func (i *avroImporter) Load(avroSpec string) (string, error) {
+	i.logger.Debugln("Load avro spec")
+
+	val, err := syntax.EvaluateExpr("",
+		fmt.Sprintf("%s%s", avro.AvroTransformerScript,
+			fmt.Sprintf("(`%s`, `%s`, `%s`)", avroSpec, i.appName, i.pkg)))
```
Comment (orlade-anz) [inline pkg/importer/avro.go:32]: Nit: nested Sprintf is hard to follow. Keep it simple.
```suggestion
	call := fmt.Sprintf("(`%s`, `%s`, `%s`)", avroSpec, i.appName, i.pkg)
	val, err := syntax.EvaluateExpr("", fmt.Sprintf("%s%s", avro.AvroTransformerScript, call))
```

File: pkg/importer/avro.go [hunk 18]
```diff
@@ -0,0 +1,49 @@
...
+import (
+	"fmt"
+
+	"github.com/anz-bank/sysl/pkg/importer/avro"
+
+	"github.com/arr-ai/arrai/syntax"
+	"github.com/sirupsen/logrus"
+)
+
+// NewAvroImporter returns a new avroImporter
```
Comment (orlade-anz) [inline pkg/importer/avro.go:12]: Nit: full stops at the ends of comments (and below).

File: pkg/importer/avro.go [hunk 19]
```diff
@@ -0,0 +1,49 @@
...
+
+	return val.String(), nil
+}
+
+func (i *avroImporter) WithAppName(appName string) Importer {
+	i.appName = appName
+	return i
+}
+
+// Set the package attribute of the imported app
```
Comment (orlade-anz) [inline pkg/importer/avro.go:45]: Nit: Comments should be descriptive (describing what it does), not prescriptive (telling it what to do, like a command). So "Sets" instead of "Set".

```suggestion
// Sets the package attribute of the imported app.
```

File: pkg/importer/avro/avro_util.arrai [hunk 20]
```diff
@@ -0,0 +1,143 @@
...
+;
+
+let util = (
+    : prettyAnnotations,
+    : transformType,
+    : getAnnotationS,
+    : getTypeName,
+    : getDefaultVal,
+    : combineTypes,
+    : correctName,
```
Comment (orlade-anz) [inline pkg/importer/avro/avro_util.arrai:141]: Nit: no spaces before `:bindings`.
```suggestion
    :prettyAnnotations,
    :transformType,
    :getAnnotationS,
    :getTypeName,
    :getDefaultVal,
    :combineTypes,
    :correctName,
```

File: pkg/importer/avro/avro_util.arrai [hunk 21]
```diff
@@ -0,0 +1,143 @@
...
+        ():'default="null"', 
+        (b: true): $'default="true"',
+        (b: false): $'default="false"',
+        (a: {}): $'default="[]"',
+        (a: [(:s), ...]): $'default="${default.a >> .s}"',
+        _: $'default="${default}"'
+    } 
+;
+
+let util = (
```
Comment (orlade-anz) [inline pkg/importer/avro/avro_util.arrai:134]: Nit: No need to assign a name to this tuple, just return it.
```suggestion
(
```

File: pkg/importer/avro/avro_util.arrai [hunk 22]
```diff
@@ -0,0 +1,143 @@
...
+            # it is array, map
+            cond typeName.s {
+                'array': 'sequence of ' + cond type('items') {{'name': (:s), ...}: correctName(s), (:s): correctName(s),},
+                'map': 'set of ' + cond type('items') {{'name': (:s), ...}: 'String' + correctName(s) + 'Item', (:s): correctName(s),},
+                _: correctName(type('name').s)
+            },
+    }
+;
+
+let getDefaultVal = \.
```
Comment (orlade-anz) [inline pkg/importer/avro/avro_util.arrai:121]: Nit: Avoid verbs in function names for simple transformations. This would read better as just `defaultVal`, since that's what it returns. What it does (i.e. `get`) is not interesting in the context of the calling code.

Also avoid using `\.` in most cases. It works, but you're passing up an opportunity to document the data you're working with. What kind of input is this function expecting? kinds of thing have a "defaultVal"? I have to be familiar with Avro to know that, and I shouldn't need to know that to understand your code.
```suggestion
let defaultVal = \.
```

File: pkg/importer/avro/avro_util.arrai [hunk 23]
```diff
@@ -0,0 +1,143 @@
...
+    }
+;
+
+let isFirstLetterUppercase = \str
+    let letter = str where .@ = 0;
+    //str.upper(letter) = letter
+;
+
+let correctName = \name
+    cond transformPrimitiveType(name) {
```
Comment (orlade-anz) [inline pkg/importer/avro/avro_util.arrai:56]: Nit: Group this function with `transformPrimitiveType`, since it appears to be the only user.

File: pkg/importer/avro/avro_util.arrai [hunk 24]
```diff
@@ -0,0 +1,143 @@
...
+# Transform Avro primitive types to Sysl ones
+let transformPrimitiveType = \type
+    cond type {
+        'null': 'null',
+        'boolean': 'bool',
+        'int': 'Int32',
+        'long': 'Int64',
+        'float': 'Float32',
+        'double': 'Float64',
+        'bytes': "bytes",
```
Comment (orlade-anz) [inline pkg/importer/avro/avro_util.arrai:28]: Nit: consistent quotes
```suggestion
        'bytes': 'bytes',
```

File: pkg/importer/avro/avro_util.arrai [hunk 25]
```diff
@@ -0,0 +1,143 @@
...
+    let annotationTxt = 
+        cond annotation {
+            "":"", 
+            _: $'${cond name {'doc':'description', _:name}}="${annotation}"'
+        };
+    annotationTxt
+;
+
+# Pretty annotation array
+let prettyAnnotations = \annotationArray
```
Comment (orlade-anz) [inline pkg/importer/avro/avro_util.arrai:14]: Nit: simpler name, same information
```suggestion
let prettyAnnotations = \annotations
```

File: pkg/importer/avro/avro_util.arrai [hunk 26]
```diff
@@ -0,0 +1,143 @@
...
+    let annotation = item(name)? .s:"";
+    let annotationTxt = 
+        cond annotation {
+            "":"", 
+            _: $'${cond name {'doc':'description', _:name}}="${annotation}"'
+        };
+    annotationTxt
+;
+
+# Pretty annotation array
```
Comment (orlade-anz) [inline pkg/importer/avro/avro_util.arrai:13]: What does this mean? Provides no more information than the function name

**Review by orlade-anz — CHANGES_REQUESTED** [review]

File: pkg/importer/avro/to_sysl_enum.arrai [hunk 27]
```diff
@@ -0,0 +1,16 @@
+# Transform Avro enum to Sysl enum
```
Comment (orlade-anz) [inline pkg/importer/avro/to_sysl_enum.arrai:1]:
```suggestion
# Transforms an Avro enum to a Sysl enum.
```

File: pkg/importer/avro/to_sysl_type.arrai [hunk 28]
```diff
@@ -0,0 +1,97 @@
+# Transform Avro record to Sysl record
```
Comment (orlade-anz) [inline pkg/importer/avro/to_sysl_type.arrai:1]:
```suggestion
# Transforms an Avro record to a Sysl type.
```

File: pkg/importer/avro/to_sysl_type.arrai [hunk 29]
```diff
@@ -0,0 +1,97 @@
...
+    [
+        buildTypeAnnotation(.('type')),
+        util.getAnnotationS(., 'doc'),
+        util.getDefaultVal(.),
+        util.getAnnotationS(., 'order'),
+    ]
+;
+
+let printMapType = \record
+$`!type String${util.transformType(record)}Item ${util.prettyAnnotations(['json_map_key="key"', util.getDefaultVal(record)])}:
```
Comment (orlade-anz) [inline pkg/importer/avro/to_sysl_type.arrai:55]: Ditto, indenting.

File: pkg/importer/avro/to_sysl_type.arrai [hunk 30]
```diff
@@ -0,0 +1,97 @@
...
+        }`::\n}
+${
+let aliases = record('aliases')? .a:[];
+cond aliases {
+    [...]: 
+$`
+
+${aliases >> ($`!alias ${.s}:
+    ${util.correctName(record('name').s)}`)::\i\n}`,
+}}`}}`;transformer
```
Comment (orlade-anz) [inline pkg/importer/avro/to_sysl_type.arrai:97]: Nit: leave some whitespace before returning.
```suggestion
}}`}}`;

transformer
```

File: pkg/importer/avro/transformer.arrai [hunk 31]
```diff
@@ -0,0 +1,105 @@
+let util = //{./avro_util};
+
+# Extra items transformed to Sysl type !enum
```
Comment (orlade-anz) [inline pkg/importer/avro/transformer.arrai:3]: I'm confused as to why these are "extra". What does "extra" mean in this context?

File: pkg/importer/avro/to_sysl_union.arrai [hunk 32]
```diff
@@ -0,0 +1,21 @@
...
+    let types = union >> util.getTypeName(.) >> .s;
+    cond {
+        //seq.contains(['null'], types):
+            cond {
+                types count > 2: printUnion(union),
+            }
+        ,
+        _: printUnion(union)
+    }
+;transformer
```
Comment (orlade-anz) [inline pkg/importer/avro/to_sysl_union.arrai:21]:
```suggestion
;

transformer
```

File: pkg/importer/avro/transformer.arrai [hunk 33]
```diff
@@ -0,0 +1,105 @@
...
+    # Types
+    ${extraTypes(schema) => //{./to_sysl_type}(.) orderby . ::\n\n}
+    # Aliases
+    ${extraAliases(schema) => //{./to_sysl_alias}(.) orderby . ::\n}
+    # Unions
+    ${extraUnions(schema) => //{./to_sysl_union}(.) orderby . ::\n\n}
+    # Enums
+    ${extraEnums(schema) => //{./to_sysl_enum}(.) orderby . ::\n}
+`;
+avroTransformer
```
Comment (orlade-anz) [inline pkg/importer/avro/transformer.arrai:105]:
```suggestion

avroTransformer
```

File: pkg/importer/avro/transformer.arrai [hunk 34]
```diff
@@ -0,0 +1,105 @@
...
+  # Load Avro spec whose format is protobuf 
+  let schema = //encoding.json.decode(schema);
+
+$`
+##########################################
+##                                      ##
+##  AUTOGENERATED CODE -- DO NOT EDIT!  ##
+##                                      ##
+##########################################
+${appName}${util.prettyAnnotations(['avro_spec="1.0"',$'${cond packageName {'':'', _:$'package="${packageName}"'}}'])}:
```
Comment (orlade-anz) [inline pkg/importer/avro/transformer.arrai:95]:
```suggestion
${appName}${util.prettyAnnotations(['avro_spec="1.0"', $`${cond packageName {'':'', _:$`package="${packageName}"`}}`])}:
```

File: pkg/importer/avro/transformer.arrai [hunk 35]
```diff
@@ -0,0 +1,105 @@
...
+    # Types
+    ${extraTypes(schema) => //{./to_sysl_type}(.) orderby . ::\n\n}
+    # Aliases
+    ${extraAliases(schema) => //{./to_sysl_alias}(.) orderby . ::\n}
+    # Unions
+    ${extraUnions(schema) => //{./to_sysl_union}(.) orderby . ::\n\n}
+    # Enums
+    ${extraEnums(schema) => //{./to_sysl_enum}(.) orderby . ::\n}
```
Comment (orlade-anz) [inline pkg/importer/avro/transformer.arrai:103]: Nit: newlines should be consistent.
```suggestion
    ${extraTypes(schema) => //{./to_sysl_type}(.) orderby . ::\n}

    # Aliases
    ${extraAliases(schema) => //{./to_sysl_alias}(.) orderby . ::\n}

    # Unions
    ${extraUnions(schema) => //{./to_sysl_union}(.) orderby . ::\n}

    # Enums
    ${extraEnums(schema) => //{./to_sysl_enum}(.) orderby . ::\n}
```

File: pkg/importer/formats.go [hunk 36]
```diff
@@ -33,6 +33,12 @@ var Grammar = Format{
 	FileExt:   []string{".g"},
 }
 
+var AVRO = Format{
```
Comment (orlade-anz) [inline pkg/importer/formats.go:36]:
```suggestion
var Avro = Format{
```

**Review by orlade-anz — CHANGES_REQUESTED** [review]
More nits, mostly around whitespace :)

File: pkg/importer/avro.go [hunk 37]
```diff
@@ -0,0 +1,49 @@
...
+	appName string
+	pkg     string
+	logger  *logrus.Logger
+}
+
+// Load returns Sysl result
+func (i *avroImporter) Load(avroSpec string) (string, error) {
+	i.logger.Debugln("Load avro spec")
+
+	val, err := syntax.EvaluateExpr("",
```
Comment (anzdaddy) [inline pkg/importer/avro.go:30]: Building parameters as strings is fragile. We should define a call interface that takes regular Go types and converts them to appropriate arr.ai types and passes them as parameters. It should end up something like:

```
rel.Call(syntax.EvaluateExpr(avro.AvroTransformerScript), avroSpec, i.appName, i.pkg)
```

File: pkg/importer/avro/tests/complex_test.arrai (hunk 2 above)
Comment (anzdaddy) [inline pkg/importer/avro/tests/complex_test.arrai]: Why does this type exist? It isn't used anywhere.

**Comment by orlade-anz** [comment]
Merged internally to be pushed out.
